ts/store/reducers/backendStatus.ts

```diff
@@ -424,6 +425,21 @@ export const isNewPaymentSectionEnabledSelector = createSelector(
     )
 );
 
+// This selector checks that both the new wallet section and the
+// new document scan section are included in the tab bar.
+// In this case, the navigation to the profile section in the tab bar
+// is replaced with the 'settings' section accessed by clicking
+// on the icon in the headers of the top-level screens.
+// It will be possible to delete this control and all the code it carries
+// it carries when isNewPaymentSectionEnabledSelector and
+// showBarcodeScanSection will be deleted
+export const isSettingsVisibleAndHideProfileSelector = createSelector(
+  isNewPaymentSectionEnabledSelector,
+  () => showBarcodeScanSection,
+  (isNewPaymentSectionEnabled, showBarcodeScan) =>
+    isNewPaymentSectionEnabled && showBarcodeScan
+);
```

Comment:
```suggestion
export const isSettingsVisibleAndHideProfileSelector = createSelector(
  isNewPaymentSectionEnabledSelector,
  (isNewPaymentSectionEnabled) =>
    isNewPaymentSectionEnabled && showBarcodeScanSection
);
```
Since `showBarcodeScanSection` comes from environment variables and its value is static, I think you can use it directly without simulating a selector in the `createSelector`. What do you think?